### PR TITLE
Add rk750 keyboard

### PR DESCRIPTION
**Pull request by yangzheng20003**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rk/rk750/config.h
Comment (zvecr) [inline keyboards/rk/rk750/config.h:1]: Please remove this file

File: keyboards/rk/rk750/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    "maintainer": "yangzheng20003",
+    "bootloader": "wb32-dfu",
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+    },
```
Comment (zvecr) [inline keyboards/rk/rk750/keyboard.json:13]:
```suggestion
    },
    "dynamic_keymap": {
        "layer_count": 3
    },
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rk/rk750/rk750.c [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2024 yangzheng20003 (@yangzheng20003)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+#define LED_ENABLE_PIN A5
+void keyboard_pre_init_kb(void) {
+    gpio_set_pin_output(LED_ENABLE_PIN);
+    gpio_write_pin_high(LED_ENABLE_PIN);
```
Comment (waffle87) [inline keyboards/rk/rk750/rk750.c:8]: What exactly is this for? All the results I'm seeing for this keyboard on Google indicate that this is a wireless keyboard. If this is initialisation code for a wireless module, this PR will not be accepted without the accompanying wireless code.

**Comment by tzarc** [comment]
This PR is on hold until #24085 is rectified.

**Comment by tzarc** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

File: keyboards/rk/rk750/rk750.c (hunk 2 above)
Comment (refineryBeans) [reply keyboards/rk/rk750/rk750.c:8]: I know I'm late but the keyboard has both a wireless and wired model, so perhaps this is for the wired model?

**Pull request closed without merging**